### PR TITLE
Add toString()

### PR DESCRIPTION
**Pull request by kaicataldo**

### CODE REVIEW

**Comment by kaicataldo** [comment]
Not sure why hub thought that was what I wanted for the PR description

File: index.js [hunk 1]
```diff
@@ -42,6 +42,16 @@ define(['nbd/Class', 'nbd/util/extend', './src/Engine'], function(Class, extend,
       this.engine.process(data);
     },
 
+    /**
+     * Generate string of styles resulting from running
```
Comment (hzoo) [inline index.js]: `Generate a string of styles as the result from` maybe?

Comment (kaicataldo) [inline index.js]: How do you feel about `Generate a string of styles as the result of`?

File: src/Engine.js [hunk 2]
```diff
@@ -78,6 +78,21 @@ define([
       }
     },
 
+    toString: function(payload) {
```
Comment (mikesherov) [inline src/Engine.js]: this will only work in the browser. You want to avoid a call to process altogether.

File: src/Engine.js [hunk 3]
```diff
@@ -78,6 +78,21 @@ define([
...
+      var i;
+
+      this.process(payload);
+      rules = this.sheet.cssRules;
+
+      for (i = 0; i < rules.length; i++) {
+        stylesString += rules[i].cssText;
+      }
+
+      return stylesString.replace(/\s+/g, '');
```
Comment (hzoo) [inline src/Engine.js]: Not sure if this is safe (i'm not sure\)?

Ex in javscript -> `var a = 1` -> `vara=1`

Comment (kaicataldo) [inline src/Engine.js]: Should be all CSS at this point

Comment (kaicataldo) [inline src/Engine.js]: But you're right, realizing this won't work for strings with spaces in them

File: src/Engine.js (hunk 2 above)
Comment (kaicataldo) [inline src/Engine.js]: Updated

File: index.js (hunk 1 above)
Comment (mikesherov) [inline index.js]: `Generates CSS derived from the data applied to the current ruleset.`

File: src/Engine.js [hunk 4]
```diff
@@ -78,6 +78,21 @@ define([
...
+    toString: function(payload) {
+      var result = '';
+      var i;
+
+      this._state = payload;
+      this._regenerateArrayRules();
+
+      for (i = 0; i < this.rules.length; i++) {
+        this.rules[i].process(this._getStateWithToggles(), this.extensions);
+        result += css.toString(this.cssRules[i], this.rules[i]);
```
Comment (mikesherov) [inline src/Engine.js]: Still using the Dom friend.

File: util/css.js [hunk 5]
```diff
@@ -108,6 +108,37 @@ define(function() {
...
+        // Literal value
+        if (isNaN(value) || isUnitlessNumber[key] || value === 0) {
+          result += key + ':' + value + ';';
+          continue;
+        }
+
+        // Numbers without units
+        result += key + ':' + value + 'px;';
+      }
+      result + '}';
```
Comment (mikesherov) [inline util/css.js]: `result +=`

**Comment by kaicataldo** [comment]
These tests will pass after https://github.com/behance/focss/pull/32 is merged in. Thanks @mikesherov!

**Comment by mikesherov** [comment]
👍

**Comment by mmase** [comment]
LGTM

**Pull request merged**